proposals/0002-gep.md

```diff
@@ -0,0 +1,207 @@
+---
+MEP: 0002
+Title: Unifed API
+Discussion: Link
+Implementation: Link
+---
+
+# TITLE: Unified API
+
+## Abstract
+
+This GEP discusses the unified API that handles the interaction with LLM providers.
+
+## Motivation
+
+[TBU, what problem we are trying to solve, focus on users, use cases, and all important context]
+
+The Unified API holds all the logic needed to create a request to a provider. This solution allows the clients to be easily
+envoked by a provider pool.
+
+
+### Requirements
+
+- R1: User should be able to specify specific models or select auto mode (selection handled by Glide)
+- R2: Should be a Hertz Client
+- R3: Should take a user request as input
+- R4: Handles all provider specific logic
+- R5: Easily maintained
+
+
+## Design
+
+
+- A Hertz client that handles an incoming request and returns a reponse from that provider.
+- Provider specific clients
+
+### Routes
+```yaml
+routes:
+    chat: api/v1/chat
+    complete: /api/v1/complete #this is also know as generate in some providers
+
+    # want to avoid specifying a provider pool as in the future we may
```

Comment:
@mkrueger12 could you please elaborate this?

In the service config GEP, I propose to have:
- multiple provider pools
- load balancing is configured on the pool level
- the pool has a unique name/ID. Using that name, it could be accessed to handle the request

Is this something you are in favor of or now?

Reference:

- https://github.com/modelgateway/geps/pull/2/files#diff-0e4c03c5c73029151e17a7f008f0e7a86d45cc14244a5567769a3929db9337f2R112-R126